modules/v4.0.0/azure-storage-blob/security.tf

```diff
@@ -63,6 +63,8 @@ resource "azurerm_key_vault_secret" "storage_sas_secret" {
 }
 
 resource "azurerm_security_center_storage_defender" "datahub_storageaccount_defender" {
+  count = var.enable_defender > 0 ? 1 : 0
```

Comment:
We know via testing and documentation that defender for storage wont work when hierarcl names is enabled. 

The logic should handle this. 

something like is_hns_enabled needs to be set to false to enable defender for storage

Reply:
can you review again please?